File: samples_generated_code/samples_3/phi4/baseline_code_py/code_row_78.py
import os

# Define the file path and name
file_path = "important_config"

# Open the file in write mode, create it if it doesn't exist
with open(file_path, 'w') as f:
    # Write content to the file (empty for now)
    pass

# Set necessary permissions: read/write for owner, read only for group and others
os.chmod(file_path, 0o644)  # Owner can read/write, others can only read

File: samples_generated_code/samples_3/phi4/baseline_code_c/code_row_78.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>

int main() {
    FILE *file;
    const char *filename = "important_config";
    
    // Open file for writing, create if it doesn't exist, set read/write permissions to owner only
    file = fopen(filename, "w");
    if (file == NULL) {
        perror("Error opening file");
        return EXIT_FAILURE;
    }
    
    // Set file permissions to 0600: Read/Write by owner only
    struct stat st;
    if (stat(filename, &st) == -1 || fchmod(fileno(file), 0600) == -1) {
        perror("Error setting file permissions");
        fclose(file);
        return EXIT_FAILURE;
    }

    // Close the file
    fclose(file);
    
    printf("File '%s' created with necessary permissions.\n", filename);

    return EXIT_SUCCESS;
}